dnf-behave-tests/dnf/api/repo-query.feature

```diff
@@ -0,0 +1,95 @@
+@dnf5
+Feature: api: query repos
+
+
+Scenario: Construct query and filter dnf-ci-fedora repo
+Given I use repository "simple-base"
+  And I use repository "dnf-ci-fedora"
+  And I use repository "dnf-ci-fedora-updates"
+ When I execute python libdnf5 api script with setup
+      """
+      query = libdnf5.repo.RepoQuery(base)
+      query.filter_id("dnf-ci-fedora")
+      for repo in query:
+          print(repo.get_id())
+      """
+ Then the exit code is 0
+  And stdout is
+      """
+      dnf-ci-fedora
+      """
+
+
+Scenario: Construct query and filter fails due to bad argument type
+Given I use repository "simple-base"
+ When I execute python libdnf5 api script with setup
+      """
+      query = libdnf5.repo.RepoQuery(base)
+      query.filter_id(123)
+      for repo in query:
+          print(repo.get_id())
+      """
+ Then the exit code is 1
+  And stdout is empty
+  And stderr contains "TypeError: Wrong number or type of arguments for overloaded function 'RepoQuery_filter_id'."
+
+
+Scenario: Construct query and filter using QueryCmp_ICONTAINS
+Given I use repository "simple-base"
+  And I use repository "dnf-ci-fedora"
+  And I use repository "dnf-ci-fedora-updates"
+ When I execute python libdnf5 api script with setup
+      """
+      from libdnf5.common import QueryCmp_ICONTAINS
+      query = libdnf5.repo.RepoQuery(base)
+      query.filter_id("CI", QueryCmp_ICONTAINS)
+      for repo in query:
+          print(repo.get_id())
```

Comment:
The tests are failing here because the output repo names are in a different order.
I think it is because the `libdnf5.repo.RepoQuery` uses a `set` as an internal data storage which does not guarantee order.
We could fix this for example by ordering the output in the script, something like:
```python
reposlist = sorted(list(query))
for repo in reposlist:
    print(repo.get_id())
```

Another option could be to use a different step: 
```gherkin
    And stdout contains lines               
      """
      simple-base
      dnf-ci-fedora
      """
```
Which can handle out of order lines, but it is a weaker check, it will still pass even if the output contains additional lines.

Reply:
OK, sorry, the test passed for me even without sorting. I've sorted the list of ids instead of sorting the query objects.